android/src/main/java/io/ably/flutter/plugin/AblyFlutterPlugin.java

```diff
@@ -65,4 +82,55 @@ public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
     private static MethodCodec createCodec() {
         return new StandardMethodCodec(new AblyMessageCodec());
     }
+
+    @Override
+    public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
+        isActivityRunning = true;
+        mainActivity = binding.getActivity();
+        binding.addOnNewIntentListener(this);
+        Intent intent = mainActivity.getIntent();
+        doIfIntentContainsRemoteMessage(intent, (message) -> {
+            // Only send the RemoteMessage to the dart side if it was actually a RemoteMessage.
+            methodCallHandler.setRemoteMessageFromUserTapLaunchesApp(message);
+        });
+    }
+
+    @Override
+    public void onDetachedFromActivityForConfigChanges() {
+    }
+
+    @Override
+    public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding binding) {
+        mainActivity = binding.getActivity();
+        binding.addOnNewIntentListener(this);
+    }
+
+    // This method does not get called when the app goes into the background
+    @Override
+    public void onDetachedFromActivity() {
+        isActivityRunning = false;
+    }
+
+    @Override
+    public boolean onNewIntent(Intent intent) {
+        mainActivity.setIntent(intent);
+        doIfIntentContainsRemoteMessage(intent, (message) -> {
+            methodChannel.invokeMethod(PlatformConstants.PlatformMethod.pushOnNotificationTap, message);
+        });
+        return false;
+    }
+
+    private Boolean doIfIntentContainsRemoteMessage(Intent intent, CompletionHandlerWithRemoteMessage completionHandler) {
```

Comment:
I was conflicted about this name as well, but settled on it because I didn't want to write `if (extras == null)` in multiple locations and wanted to encapsulate the condition. Do you have a suggestion for the name?

Reply:
This was renamed to `handleRemoteMessageIntent`, in 1ddee9b3edde22bc1c28451cfced568f49b8a2e1.